lightning/src/offers/invoice_request.rs

```diff
@@ -1131,13 +1131,15 @@ impl TryFrom<PartialInvoiceRequestTlvStream> for InvoiceRequestContents {
 		}
 
 		offer.check_quantity(quantity)?;
-
-		let amount_msats = match offer.amount() {
-			None => amount,
-			Some(Amount::Bitcoin { amount_msats }) => Some(amount_msats),
-			Some(Amount::Currency { iso4217_code: _ , amount: _ }) => {
-				return Err(Bolt12SemanticError::UnsupportedCurrency)
-			},
+		let amount_msats = match amount {
+			Some(amount) => Some(amount),
+			None => match offer.amount() {
+				Some(Amount::Bitcoin { amount_msats }) => Some(amount_msats),
```

Comment:
This would cause us to set the amount when the parsed `InvoiceRequest` didn't have one. I think we only need to call `check_amount_msats_for_quantity` when we have `Some(amount)`.

Reply:
I noticed that calling `check_amount_msats_for_quantity` only when we have `Some(amount)` causes the test `parses_invoice_request_with_amount` to fail. It seems we need to call it even when amount is `None`. Skipping the `None` check misses the validation for the offer's currency, so the test doesn't catch the `UnsupportedCurrency` error.

So for now, I kept the call to `check_amount_msats_for_quantity` regardless of whether amount is Some or None. Let me know if this approach is correct!

Reply:
Right, we should still call that, but as written my earlier point still stands. We are now setting `Some(amount_msats)` on the `InvoiceRequestContents` below when we shouldn't.